ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -1831,6 +1831,10 @@ protected synchronized final void phase2Commit (boolean reportHeuristics) throws
 
             phase2Abort(reportHeuristics);
         }
+        else if ((preparedList == null || preparedList.size() == 0) && actionStatus == ActionStatus.ABORTED) {
+            // the prepared list was completely processed but some resource rolled back
+            phase2Abort(reportHeuristics);
```

Comment:
This is a scenario where one of the resources rolls back during the prepare phase but there are no heuristics which I think is just part of the normal behaviour of committing resources. A similar situation occurs when the first resource aborts during commit so we abort the remaining participants but we don't log a warning in this case. [Here is where we abort the remaining participants](https://github.com/jbosstm/narayana/blob/master/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java#L2829)

If you still think such a scenario is sufficiently noteworthy then I will add a WARN message.

On the other hand the pending list means that there are resources for which prepare was never called and this is  unusual so is reported with a warning.